cmd/horcrux/cmd/metrics.go

```diff
@@ -0,0 +1,83 @@
+package cmd
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"os"
+	"time"
+
+	"github.com/armon/go-metrics"
+	gmprometheus "github.com/armon/go-metrics/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	tmlog "github.com/tendermint/tendermint/libs/log"
+)
+
+func AddPrometheusMetrics(mux *http.ServeMux) {
+	logger := tmlog.NewTMLogger(tmlog.NewSyncWriter(os.Stdout)).With("module", "metrics")
+
+	// Add raft metrics to prometheus
+	enableRaftMetrics := true
+	if enableRaftMetrics {
+		// PrometheusSink config w/ definitions for each metric type
+		cfg := gmprometheus.DefaultPrometheusOpts
+		sink, err := gmprometheus.NewPrometheusSinkFrom(cfg)
+		if err != nil {
+			logger.Error("Could not configure Raft Metrics")
+		}
+		defer prometheus.Unregister(sink)
```

Comment:
It's safe to unregister the sink at the conclusion of this function?

Reply:
Better not to unregister :)